samples/sample_common/src/sample_utils.cpp

```diff
@@ -180,16 +180,27 @@ mfxStatus CSmplYUVReader::LoadNextFrame(mfxFrameSurface1* pSurface)
 
     mfxU32 nBytesPerPixel = (pInfo.FourCC == MFX_FOURCC_P010 || pInfo.FourCC == MFX_FOURCC_P210 ) ? 2 : 1;
 
-    if (MFX_FOURCC_YUY2 == pInfo.FourCC || MFX_FOURCC_RGB4 == pInfo.FourCC || MFX_FOURCC_BGR4 == pInfo.FourCC || pInfo.FourCC == MFX_FOURCC_Y210 || pInfo.FourCC == MFX_FOURCC_Y410)
+    if (   MFX_FOURCC_YUY2 == pInfo.FourCC
+        || MFX_FOURCC_RGB4 == pInfo.FourCC
+        || MFX_FOURCC_BGR4 == pInfo.FourCC
+        || MFX_FOURCC_AYUV == pInfo.FourCC
+        || MFX_FOURCC_A2RGB10 == pInfo.FourCC
+#if (MFX_VERSION >= 1027)
+        || MFX_FOURCC_Y210 == pInfo.FourCC
+        || MFX_FOURCC_Y410 == pInfo.FourCC
+#endif
+    )
     {
         //Packed format: Luminance and chrominance are on the same plane
         switch (m_ColorFormat)
         {
+        case MFX_FOURCC_A2RGB10:
+        case MFX_FOURCC_AYUV:
         case MFX_FOURCC_RGB4:
         case MFX_FOURCC_BGR4:
             pitch = pData.Pitch;
             ptr = MSDK_MIN( MSDK_MIN(pData.R, pData.G), pData.B);
-            ptr = ptr + pInfo.CropX + pInfo.CropY * pData.Pitch;
+            ptr = ptr + pInfo.CropX*4 + pInfo.CropY * pData.Pitch;
```

Comment:
Hm. Did we have a bug here handling non-zero CropX? Looks like yes. And this should be a _separate_ commit.

Also, please, use '+=' to avoid ptr on both sides of the equation:
`ptr += pInfo.CropX*4 + pInfo.CropY * pData.Pitch;`

Reply:
Yes, it was a bug. No, it won't be separated because it's a part of file reader/writer rework - one of major tasks of this commit

Reply:
You have 3 commits, not 1! And changes in this commit should be distributed across another 2 per my perspective. I will not approve the change without that.